### ISSUE TITLE
Composite primary key

### ISSUE THREAD

**Issue by Nic0tiN** [vierge-noire/cakephp-fixture-factories#26]
Hello,

I have a schema with a composite primary key. When calling `BaseFactory::persist()`, DataCompiler returns a TypeError `DataCompiler::getRootTablePrimaryKey() must be a string, array returned`.

`DataCompiler::getRootTablePrimaryKey()` return type is set to `string`
```php
    public function getRootTablePrimaryKey(): string
    {
        return $this->getFactory()->getRootTableRegistry()->getPrimaryKey();
    }
```

and is therefore incompatible with `Cake\ORM\Table::getPrimaryKey()` which returns `string|string[]`

**Comment by pabloelcolombiano** [comment]
Thanks for the hint.

I'll update the package asap.

**Comment by pabloelcolombiano** [comment]
This should now be fixed with versions v1.0.1 (CakePHP 3.x) and v2.0.1 (CakePHP 4.x)

Thanks for the past and future feedbacks.
